dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -228,7 +229,8 @@ Entry updateEntryMetadata(Entry entry, AbstractEntryClient.Type type) {
 
             // Find filepath to parse
             for (Tag tag : ((Tool)entry).getVersions()) {
```

Comment:
If I understand correctly, this is basically second-guessing getMainBranch, making sure that we check git references or Docker tags. Is it possible just to fix getMainBranch to properly return git references always so we don't have to second guess? (Take a look to see what else is using getMainBranch and might be broken by such a change)

How do the GitLab and Bitbucket implementations behave?

Reply:
ok, please give it a shot here and if the scope seems to creep too far, then we can merge this as is to be followed up with that PR to develop